lib/spack/spack/stage.py

```diff
@@ -176,13 +176,14 @@ def _resolve_paths(candidates):
 
 # Cached stage path root
 _stage_root = None
+_default_stage_config = ["$tmpdir/$user/spack-stage", "$user_cache_path/stage"]
```

Comment:
There's a typo here:
```suggestion
_default_stage_config = ["$tempdir/$user/spack-stage", "$user_cache_path/stage"]
```
that result in literally `$tmpdir` being used:
```
$ spack -c "config::{}" stage zlib-ng
==> Using cached archive: /home/culpo/worktrees/spack/hotfixes/var/spack/cache/_source-cache/archive/ec/ec6a76169d4214e2e8b737e0850ba4acb806c69eeace6240ed4481b9f5c57cdf.tar.gz
==> Staged zlib-ng in /home/culpo/worktrees/spack/hotfixes/$tmpdir/culpo/spack-stage/spack-stage-zlib-ng-2.2.1-smijmn3pb2nbxrawr73uqtw4xbfrvnxy
```